mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java

```diff
@@ -880,12 +881,14 @@ private MarcBibCollection toMarcBibCollection(QueryResult result) {
   }
 
   @Override
-  public Future<Void> cleanRecords(String tenantId) {
+  public Future<Void> deleteRecords(String tenantId, int lastUpdatedDays, int limit) {
     Promise<Void> promise = Promise.promise();
-    var purgeOldRecordsQuery = DSL.deleteFrom(RECORDS_LB)
-      .where(RECORDS_LB.STATE.eq(RecordState.OLD).and(RECORDS_LB.MATCHED_ID.in(DSL.select(RECORDS_LB.ID).from(RECORDS_LB).where(RECORDS_LB.STATE.eq(RecordState.DELETED)))));
-    var purgeDeletedRecordsQuery = DSL.deleteFrom(RECORDS_LB)
-      .where(RECORDS_LB.STATE.eq(RecordState.DELETED));
+    var selectDeletedRecordIdsQuery = DSL
+      .select(RECORDS_LB.ID).from(RECORDS_LB)
+      .where(RECORDS_LB.STATE.eq(RecordState.DELETED)).and(RECORDS_LB.UPDATED_DATE.le(OffsetDateTime.now().minusDays(lastUpdatedDays)))
+      .limit(limit);
+    var purgeOldRecordsQuery = DSL.deleteFrom(RECORDS_LB).where(RECORDS_LB.STATE.eq(RecordState.OLD).and(RECORDS_LB.MATCHED_ID.in(selectDeletedRecordIdsQuery)));
+    var purgeDeletedRecordsQuery = DSL.deleteFrom(RECORDS_LB).where(RECORDS_LB.ID.in(selectDeletedRecordIdsQuery));
```

Comment:
As I can see from these queries `limit` parameter is not actually limit the total amount of records:
Records that will be removed if limit = 10:
 - 10 records with the DELETED state
 - from 0 to an unlimited amount of records with the OLD state

This may cause a problems